src/components/expression/_expression.scss

```diff
@@ -41,54 +42,54 @@
     border-bottom-style: solid;
     transform: translateY(-1px);
   }
+}
+
+.euiExpression__icon {
+  margin-left: $euiSizeXS;
+}
 
+.euiExpression-isActive {
+  border-bottom-style: solid;
 }
 
 .euiExpression-columns {
-  background-color: $euiColorLightestShade;
   width: 100%;
   display: flex;
-  padding: $euiSizeXS * .5;
+  padding: $euiSizeXS;
   border-radius: $euiSizeXS;
 
-  &:not(.euiExpression-isClickable) {
-    .euiExpression__description {
-      color: $euiTextColor;
+  &.euiExpression-isClickable {
+    background-color: $euiColorLightestShade;
+
+    // sass-lint:disable-block nesting-depth
+    &:focus,
+    &:hover:not(:disabled) {
+      .euiExpression__description,
+      .euiExpression__value {
+        // inner child specificity so it inherits underline color from text color
+        text-decoration: underline;
+      }
     }
   }
 
-  .euiExpression__description,
-  .euiExpression__value {
-    @include euiTextBreakWord;
-  }
-
   .euiExpression__description {
-    margin: $euiSizeXS;
     text-align: right;
+    margin-right: $euiSizeS;
+    flex-shrink: 0; // Ensures it doesn't get smaller in case the value is really long
   }
 
   .euiExpression__value {
     flex-grow: 1;
-    display: flex;
-    margin: $euiSizeXS;
   }
 
   .euiExpression__icon {
-    margin-left: auto;
+    margin-top: $euiSizeXS;
   }
 }
 
-.euiExpression-isActive {
-  border-bottom-style: solid;
-}
-
-.euiPopover__anchor {
-  width: 100%;
-}
```

Comment:
Don't forget that you can't set external component-specific style like this. It will affect **all** usages. **EuiPopover** actually has a `display="block"` prop to help with this which I added to your examples.